core/include/crypto/crypto.h

```diff
@@ -153,13 +153,15 @@ struct ecc_public_key {
 	struct bignum *x;	/* Public value x */
 	struct bignum *y;	/* Public value y */
 	uint32_t curve;	        /* Curve type */
+	const void *ops;	/* Operations linked to the key */
```

Comment:
Why not `const struct crypto_ecc_public_ops *`?

Reply:
Because of the SM2_KEP algorithm using ECC keys but is just a key derivation operation not sign/verify and SM2_PKE algorithm using also ECC keys but it's encrypt/decrypt operation.
Making `ops` as void prevent defining for all algorithms using ECC key to same operation structure. Like this:
```
struct crypto_ecc_public_ops{
   TEE_Result verify(...);
   TEE_Result encrypt(...);
   void free(...);
}
```
and

```
struct crypto_ecc_keypair_ops{
    TEE_Result generate(...);
    TEE_Result sign(...);
    TEE_Result shared_secret(...);
    TEE_Result decrypt(...);
    TEE_Result derive(...);
}
```

My idea was to modify how to operate with SM2_PKE and SM2_KEP (in another PR) if this PR is accepted by defining operations structure for each algo, like:
```
struct crypto_sm2_pke_public_ops{
   TEE_Result encrypt(...);
   void free(...);
}
```
and

```
struct crypto_sm2_pke_keypair_ops{
    TEE_Result generate(...);
    TEE_Result decrypt(...);
}
```

```
struct crypto_sm2_kep_keypair_ops{
    TEE_Result generate(...);
    TEE_Result derive(...);
}
```

Now if you want:
Option 1: I can define one operation structure for ECC public and one for ECC private key with all possible operations regardless the algorithm and add one or two commits (one for each SM2 algo KEP/PKE) in the PR to see what will be.
Option 2: I can add one or two commits (one for each SM2 algo KEP/PKE) in the PR for one operation structure for each algo and see what will be.

Let me know which option will be better for you.

Reply:
I'm not so keen on skipping the type check by the compiler. So I'd prefer option 1, but it's enough to only define what's needed right now. We can extend the ops struct later, assuming that `NULL` function pointers in the struct will mean not supported.